gmodule.lua

```diff
@@ -156,11 +155,15 @@ function gModule:share(gm, ...)
 end
 
 function gModule:training()
-	self:apply(function(module) module:training() end)
```

Comment:
Unfortunately no. `nn.Module:apply(callback)` also calls `callback(self)`; in these cases that would lead to an infinite recursion.